scripts/src/models/index.ts

```diff
@@ -13,12 +14,12 @@ let initPromise: Promise<string>;
 export type ModelConstructor = { new(): Model };
 export type ModelMemberInitializer = () => any;
 export abstract class Model extends BaseEntity {
-	public static new<T extends Model>(this: ObjectType<T>, data?: Partial<T>): T {
-		const instance = (this as typeof BaseEntity).create(data) as T;
+	public static new<T extends Model>(this: ObjectType<T>, data?: DeepPartial<T>): T {
+		const instance = (this as typeof BaseEntity).create(data!) as T;
 
 		for (const [name, initializer] of (this as typeof Model).initializers.entries()) {
```

Comment:
why not `name: keyof Model`

Reply:
there's no way to add type annotation when destructuring:
https://github.com/Microsoft/TypeScript/issues/7576